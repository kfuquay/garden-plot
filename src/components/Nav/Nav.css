nav {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: baseline;
  padding: 0.5%;
  border-bottom: 1px solid rgba(232, 165, 93, 0.4);
}

.NavLink {
  text-decoration: none;
  margin: 0 8px;
  color: hotpink;
  text-transform: uppercase;
  letter-spacing: 0.09em;
  font-weight: 300;
  font-size: 14px;
}

.header {
  flex: 2 2 40%;
  color: #bada55;
}

@media only screen and (max-device-width: 480px) {
  .menu {
    display: flex;
    flex-direction: column;
  }

  nav {
    border: none;
  }
}
