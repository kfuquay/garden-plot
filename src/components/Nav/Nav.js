import React, { Component, Fragment } from "react";
import GardenContext from "../../context/GardenContext";
import { NavLink } from "react-router-dom";
import TokenService from "../../services/token-service";
import "./Nav.css";

class Nav extends Component {
  static contextType = GardenContext;

  handleLogoutClick = () => {
    this.context.setCurrentUser("");
    TokenService.clearAuthToken();
  };

  // render logout and add project links only if user is currently logged in
  renderLogoutLink() {
    return (
      <Fragment>
        <NavLink to="/new" className="NavLink">
          <i className="fas fa-plus icon" />
        </NavLink>
        <NavLink to="/dash" className="NavLink">
          Dash
        </NavLink>
        <NavLink
          className="NavLink"
          onClick={this.handleLogoutClick}
          to="/login"
        >
          Logout
        </NavLink>
      </Fragment>
    );
  }

  //if user is not logged in, render login link
  renderLoginLink() {
    return (
      <NavLink to="/login" className="NavLink">
        Log In
      </NavLink>
    );
  }
  render() {
    return (
      <nav role="navigation">
        <NavLink to="/" className="NavLink">
          <h1 className="header">
            <i className="fas fa-carrot" />
            Garden Plot
          </h1>
        </NavLink>
        <div>
          {TokenService.hasAuthToken()
            ? this.renderLogoutLink()
            : this.renderLoginLink()}
        </div>
      </nav>
    );
  }
}

export default Nav;
