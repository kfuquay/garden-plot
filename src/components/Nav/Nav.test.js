import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Nav from "./Nav";

describe(`Nav component`, () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<Nav />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
