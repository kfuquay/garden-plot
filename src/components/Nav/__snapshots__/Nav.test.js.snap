// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders without crashing 1`] = `
<nav
  role="navigation"
>
  <NavLink
    className="NavLink"
    to="/"
  >
    <h1
      className="header"
    >
      <i
        className="fas fa-carrot"
      />
      Garden Plot
    </h1>
  </NavLink>
  <div
    className="menu"
  >
    <NavLink
      className="NavLink"
      to="/login"
    >
      Log In
    </NavLink>
  </div>
</nav>
`;
